domain/src/main/java/life/qbic/apps/datamanager/notifications/MessageParameters.java

```diff
@@ -0,0 +1,33 @@
+package life.qbic.apps.datamanager.notifications;
+
+import java.time.Instant;
+
+/**
+ * <b><class short description - 1 Line!></b>
+ *
+ * <p><More detailed description - When to use, what it solves, etc.></p>
+ *
+ * @since <version tag>
+ */
+public final class MessageParameters {
+
+  public final String messageType;
+
+  public final String messageId;
+
+  public final String occuredOn;
+
+  private MessageParameters(String messageType, String messageId, String occuredOn) {
+    super();
+    this.messageId = messageId;
+    this.messageType = messageType;
+    this.occuredOn = occuredOn;
+  }
+
+  public static MessageParameters durableTextParameters(String messageType, String messageId,
+      Instant occuredOn) {
+    String occuredOnString = occuredOn.toString();
```

Comment:
is it bad practice to do this in the constructor? no critique, just curiousity

Reply:
Basically what we learned from Joshua Bloch's Effective Java Item 1: Consider static factory methods over constructors. If you follow the newer Java API, you will find it everywhere. 

https://dev.to/kylec32/effective-java-tuesday-let-s-consider-static-factory-methods-170p